pallets/resource-discussions/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The maximum number of characters in a resource id.
+        #[pallet::constant]
+        type MaxResourceIdLength: Get<u32>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ResourceDiscussionLinked {
+            resource_id: ResourceId<T>,
+            account_id: T::AccountId,
+            post_id: PostId,
+        },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        ResourceDiscussionAlreadyCreated,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn resource_post)]
+    pub type ResourceDiscussion<T: Config> =
+        StorageDoubleMap<_, Blake2_128Concat, ResourceId<T>, Twox64Concat, T::AccountId, PostId>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(500_000_000))]
+        #[transactional]
+        pub fn link_post_to_resource(
+            origin: OriginFor<T>,
+            resource_id: ResourceId<T>,
+            post_id: PostId,
+        ) -> DispatchResult {
+            let caller = ensure_signed(origin)?;
+
+            Self::do_link_post_to_resource(caller, resource_id, post_id)
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(Weight::from_ref_time(750_000_000))]
+        #[transactional]
+        pub fn create_resource_discussion(
+            origin: OriginFor<T>,
+            resource_id: ResourceId<T>,
+            space_id: PostId,
+            content: Content,
+        ) -> DispatchResult {
+            let caller = ensure_signed(origin)?;
+
+            // No need to create discussion, use "link post to resource instead"
+            ensure!(
+                !ResourceDiscussion::<T>::contains_key(resource_id.clone(), caller.clone()),
+                Error::<T>::ResourceDiscussionAlreadyCreated,
+            );
+
+            let space = pallet_spaces::Pallet::<T>::require_space(space_id)?;
+
+            let post_id = NextPostId::<T>::get();
+
+            // This call also ensures that [caller] has permission to create posts in that space
+            pallet_posts::Pallet::<T>::create_post(
+                RawOrigin::Signed(caller.clone()).into(),
+                Some(space.id),
+                PostExtension::RegularPost,
+                content,
+            )?;
+
+            Self::do_link_post_to_resource(caller, resource_id, post_id)
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        fn do_link_post_to_resource(
+            caller: T::AccountId,
+            resource_id: ResourceId<T>,
+            post_id: PostId,
+        ) -> DispatchResult {
+            let post = pallet_posts::Pallet::<T>::require_post(post_id)?;
+            post.ensure_owner(&caller)?;
```

Comment:
Should we move these two lines to `update_resource_discussion()` (aka `link_post_to_resource()`)? Because in the case of `create_resource_discussion()` extrinsic, we already know that this `caller` account is an owner of the post, and the post exists.